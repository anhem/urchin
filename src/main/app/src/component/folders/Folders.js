import React, {Component} from "react";
import {Link} from "react-router-dom";

class Folders extends Component {

    render() {
        let folders = this.props.folders || [];
        let folderItems = folders.map((item, index) =>
            <li className="folder-list__folder"
                key={index.toString()}>
                <Link to={`/folders/${item.folderId}`}>
                    {item.folderName}
                </Link>
            </li>
        );
        return (
            <div data-view="folders"
                 className="folders">
                <Link data-view="newFolder"
                      className="folders__new-folder"
                      to="/folders/new">
                    new folder</Link>
                <h2>Folders</h2>
                <ul className="folder-list">
                    {folderItems}
                </ul>
            </div>
        )
    }
}

export default Folders