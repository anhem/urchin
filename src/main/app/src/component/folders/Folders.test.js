import React from 'react';
import {mount, shallow} from 'enzyme'
import {Router} from "react-router-dom";
import toJson from 'enzyme-to-json';
import Folders from "./Folders";
import history from '../../history'

describe('Folders', () => {

    it('renders without crashing', () => {
        let component = mount(
            <Router history={history}>
                <Folders/>
            </Router>
        );
        expect(component.length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<Folders/>))).toMatchSnapshot();
    });
});
