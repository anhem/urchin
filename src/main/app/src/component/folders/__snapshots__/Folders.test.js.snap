// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folders match snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/folders"
  />
  <Route
    component={[Function]}
    path="/folders/new"
  />
  <Route
    component={[Function]}
    path="/folders/:id"
  />
</Switch>
`;
